veps/sig-compute/10-dra-devices/vep.md

```diff
@@ -0,0 +1,760 @@
+kind: ResourceClaimTemplate
+metadata:
+  name: single-gpu
+  namespace: gpu-test1
+spec:
+  spec:
+    devices:
+      requests:
+      - allocationMode: ExactCount
+        count: 1
+        deviceClassName: vfiopci.nvidia.com
+        name: gpu
+---
+apiVersion: resource.k8s.io/v1alpha3
+kind: DeviceClass
+metadata:
+  name: vfiopci.example.com
+spec:
+  config:
+  - opaque:
+    driver: gpu.nvidia.com
+    parameters:
+      apiVersion: gpu.nvidia.com/v1alpha1
+      driverConfig:
+       driver: vfio-pci
+      kind: GpuConfig
+  selectors:
+  - cel:
+      expression: device.driver == 'gpu.nvidia.com' && device.attributes['gpu.nvidia.com'].type == 'gpu'
+```
+If the above driver is deployed in a cluster with three nodes with two different GPUs, say `RTX 4080` and `RTX 3080`.
+
+The user consumes the GPU using the following spec:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  name: vmi-fedora
+  namespace: gpu-test1
+spec:
+  resourceClaims:
+  - name: gpu-resource-claim
+    resourceClaimTemplateName: single-gpu
+  domain:
+    gpus:
+    - claim:
+         name: gpu-resource-claim
+         request: gpu
+      name: example-pgpu
+```
+
+The user will then wait for devices to be allocated. The device made available to the VMI will be available in the
+status:
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  name: vmi-fedora
+  namespace: gpu-test1
+spec:
+  resourceClaims:
+  - name: gpu-resource-claim
+    resourceClaimTemplateName: single-gpu
+  domain:
+    gpus:
+    - claim:
+         name: gpu-resource-claim
+         request: gpu
+      name: example-pgpu
+status:
+  deviceStatus:
+    gpuStatuses:
+    - name: example-pgpu
+      deviceResourceClaimStatus:
+        name: gpu-0
+        resourceClaimName: virt-launcher-vmi-fedora-hhzgn-gpu-resource-claim-c26kh
+        attributes:
+          pciAddress: 0000:01:00.0
+```
+
+## Alternatives
+
+## Alternative 1
+```
+type GPU struct {
+	// Name of the GPU device as exposed by a device plugin
+	Name              string       `json:"name"`
+	DeviceName        string       `json:"deviceName"`
+	VirtualGPUOptions *VGPUOptions `json:"virtualGPUOptions,omitempty"`
+	// If specified, the virtual network interface address and its tag will be provided to the guest via config drive
+	// +optional
+	Tag   string `json:"tag,omitempty"`
+	Claim string `json:"claim,omitempty"`
+}
+
+type HostDevice struct {
+	Name string `json:"name"`
+	// DeviceName is the resource name of the host device exposed by a device plugin
+	DeviceName string `json:"deviceName"`
+	// If specified, the virtual network interface address and its tag will be provided to the guest via config drive
+	// +optional
+	Tag string `json:"tag,omitempty"`
+	// If specified, the ResourceName of the host device will be provisioned using DRA driver . which will not require the deviceName field
+	//+optional
+	ResourceClaim *ResourceClaim `json:"resourceClaim,omitempty"`
```

Comment:
Why not:
```suggestion
	ResourceClaim *k8sv1.ResourceClaim `json:"resourceClaim,omitempty"`
```
That would effectively be the same thing as the main proposal, just without the additional struct and legacy API to maintain.

Reply:
this is implementation detail, is it okay if we discuss it during code-review?